android/src/main/java/com/pkmnapps/nearby_connections/LocationEnabler.java

```diff
@@ -72,49 +66,45 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
 
     private void initiateLocationServiceRequest() {
         LocationRequest mLocationRequest = LocationRequest.create();
-        LocationSettingsRequest.Builder builder = new LocationSettingsRequest.Builder();
-        builder.addLocationRequest(mLocationRequest);
+        LocationSettingsRequest.Builder builder = new LocationSettingsRequest
+                .Builder()
+                .addLocationRequest(mLocationRequest)
+                .setAlwaysShow(true);
         mLocationSettingsRequest = builder.build();
     }
 
-    private boolean checkLocationService() {
-        boolean gps_enabled = mLocationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
-        boolean network_enabled = mLocationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER);
-        return gps_enabled || network_enabled;
-    }
-
     void requestLocationEnable(final Result result) {
-        try {
-            if (this.checkLocationService()) {
-                result.success(true);
-                return;
-            }
-        } catch (Exception e) {
-            result.error("LOCATION_SERVICE_ERROR", "Unable to determine location service status", null);
-            return;
-        }
-
         this.pendingResult = result;
-        mSettingsClient.checkLocationSettings(mLocationSettingsRequest).addOnFailureListener(activity,
-                new OnFailureListener() {
-                    @Override
-                    public void onFailure(@NonNull Exception e) {
-                        if (e instanceof ResolvableApiException) {
-                            ResolvableApiException resolvableApiException = (ResolvableApiException) e;
-                            int statusCode = resolvableApiException.getStatusCode();
-                            if (statusCode == LocationSettingsStatusCodes.RESOLUTION_REQUIRED) {
-                                try {
-                                    resolvableApiException.startResolutionForResult(activity, GPS_ENABLE_REQUEST);
-                                } catch (IntentSender.SendIntentException sie) {
-                                    result.error("LOCATION_SERVICE_ERROR", "Unable to resolve location request",
-                                            null);
-                                }
+        Task<LocationSettingsResponse> task = LocationServices.getSettingsClient(activity)
+                .checkLocationSettings(mLocationSettingsRequest);
+
+        task.addOnCompleteListener(new OnCompleteListener<LocationSettingsResponse>() {
```

Comment:
Hi @mannprerak2, Is this you were talking aboout?

Reply:
Yeah this seems fine